src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,549 @@
+const timeHelpText = `(journalctl date format, e.g. YYYY-MM-DD)`
+
+type step func() error
+
+type stepParams struct {
+	fs            afero.Fs
+	m             *sync.Mutex
+	w             *zip.Writer
+	timeout       time.Duration
+	producedFiles chan string
+}
+
+// Creates a file in the zip writer with name 'filename' and writes 'contents' to it.
+func writeFileToZip(ps *stepParams, filename string, contents []byte) error {
+	ps.m.Lock()
+	defer ps.m.Unlock()
+
+	wr, err := ps.w.Create(filename)
+	if err != nil {
+		return err
+	}
+	_, err = wr.Write(contents)
+	if err != nil {
+		return fmt.Errorf("couldn't save '%s': %w", filename, err)
+	}
+	ps.producedFiles <- filename
+	return nil
+}
+
+// Runs a command and pipes its output to a new file in the zip writer.
+func writeCommandOutputToZip(
+	ps *stepParams, filename string, command string, args ...string,
+) error {
+	ps.m.Lock()
+	defer ps.m.Unlock()
+
+	ctx, _ := context.WithTimeout(context.Background(), ps.timeout)
+	cmd := exec.CommandContext(ctx, command, args...)
+
+	wr, err := ps.w.Create(filename)
+	if err != nil {
+		return err
+	}
+	cmd.Stdout = wr
+
+	err = cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		return fmt.Errorf("couldn't save '%s': %w", filename, err)
+	}
+	ps.producedFiles <- filename
+	return nil
+
+}
+
+func NewBundleCommand(fs afero.Fs, mgr config.Manager) *cobra.Command {
+	var (
+		brokers        []string
+		configFile     string
+		user           string
+		password       string
+		mechanism      string
+		enableTLS      bool
+		certFile       string
+		keyFile        string
+		truststoreFile string
+		logsSince      string
+		logsUntil      string
+		timeout        time.Duration
+	)
+	command := &cobra.Command{
+		Use:   "bundle",
+		Short: "Collect environment data and create a bundle file for the Vectorized support team to inspect.",
+		Long: `'rpk debug bundle' collects environment data that can help debug and diagnose
+issues with a redpanda cluster, a broker, or the machine it's running on. It
+then bundles the collected data into a zip file.
+
+The following are the data sources that are bundled in the compressed file:
+
+ - Kafka data: Number of brokers, consumer groups, topics & partitions, topic
+  configuration and replication factor.
+ - /proc/cpuinfo: CPU information like make, core count, cache, frequency.
+ - /proc/interrupts: IRQ distribution across CPU cores.
+ - redpanda logs: The redpanda logs written to journald. If --logs-since or
+  --logs-until are passed, then only the logs within the resulting time frame
+  will be included.
+ - redpanda configuration: The redpanda configuration file (redpanda.yaml).
+ - Resource usage data: CPU usage percentage, free memory available for the
+  redpanda process.
+ - Clock drift: The ntp drift as reported by 'clockdiff'. Needs CAP_NET_RAW,
+  so it's only included if 'rpk debug bundle' is run as root.
+ - Network config: As reported by 'ifconfig'.
```

Comment:
any preference for ifconfig vs ip?  ifconfig has also been deprecated so perhaps might be good just to go with ip only.

Reply:
I personally use `ip` but during our conversation you mentioned `if`, so I used that one instead. I might have been a typo, then. However, I'll be trying to replace all the cmds with Go code/ existing libs, so I don't think I'll use either.

Reply:
Fixed.